server.go

```diff
@@ -255,12 +255,16 @@ func (s *ExtensionManagerServer) Run() error {
 		for {
 			time.Sleep(s.pingInterval)
 
+			s.mutex.Lock()
+			serverClient := s.serverClient
+			s.mutex.Unlock()
```

Comment:
I'm not totally sure I understand why this helps. I think it's so that this routine can happen at the same time some other thread might have called shutdown (and set `s.serverClient` to nil), but I'm not totally sure I see it. 

Is this about a narrow race where shutdown happens between `if serverClient == nil ` and `s.serverClient.Ping()` ?

Reply:
Yes, it's narrow.

Here's the race condition if we don't add the locks (it's between the read `serverClient := s.serverClient` and the write `s.serverClient = nil` in `Shutdown`):
```
=== RUN   TestNoDeadlockOnError
==================
WARNING: DATA RACE
Write at 0x00c00010f610 by goroutine 14:
  github.com/osquery/osquery-go.(*ExtensionManagerServer).Shutdown()
      /Users/luk/fleetdm/git/forks/osquery-go/server.go:352 +0x3d2
  github.com/osquery/osquery-go.(*ExtensionManagerServer).Run()
      /Users/luk/fleetdm/git/forks/osquery-go/server.go:279 +0x1d0
  github.com/osquery/osquery-go.TestNoDeadlockOnError()
      /Users/luk/fleetdm/git/forks/osquery-go/server_test.go:57 +0x55c
  testing.tRunner()
      /Users/luk/go/src/testing/testing.go:1595 +0x238
  testing.(*T).Run.func1()
      /Users/luk/go/src/testing/testing.go:1648 +0x44

Previous read at 0x00c00010f610 by goroutine 16:
  github.com/osquery/osquery-go.(*ExtensionManagerServer).Run.func2()
      /Users/luk/fleetdm/git/forks/osquery-go/server.go:259 +0x64

Goroutine 14 (running) created at:
  testing.(*T).Run()
      /Users/luk/go/src/testing/testing.go:1648 +0x82a
  testing.runTests.func1()
      /Users/luk/go/src/testing/testing.go:2054 +0x84
  testing.tRunner()
      /Users/luk/go/src/testing/testing.go:1595 +0x238
  testing.runTests()
      /Users/luk/go/src/testing/testing.go:2052 +0x896
  testing.(*M).Run()
      /Users/luk/go/src/testing/testing.go:1925 +0xb57
  main.main()
      _testmain.go:69 +0x2bd

Goroutine 16 (running) created at:
  github.com/osquery/osquery-go.(*ExtensionManagerServer).Run()
      /Users/luk/fleetdm/git/forks/osquery-go/server.go:255 +0x18c
  github.com/osquery/osquery-go.TestNoDeadlockOnError()
      /Users/luk/fleetdm/git/forks/osquery-go/server_test.go:57 +0x55c
  testing.tRunner()
      /Users/luk/go/src/testing/testing.go:1595 +0x238
  testing.(*T).Run.func1()
      /Users/luk/go/src/testing/testing.go:1648 +0x44
==================
    testing.go:1465: race detected during execution of test
--- FAIL: TestNoDeadlockOnError (0.00s)
```